pkg/cluster/internal/kubeadm/config.go

```diff
@@ -319,6 +287,24 @@ metadata:
   name: config
 `
 
+// JoinConfigTemplate is the kubadm join config template
+const JoinConfigTemplate = `# config generated by kind
+apiVersion: kubeadm.k8s.io/{{ .KubeadmAPI }}
+kind: JoinConfiguration
+metadata:
+  name: config
+nodeRegistration:
+  criSocket: "/run/containerd/containerd.sock"
+  kubeletExtraArgs:
+    fail-swap-on: "false"
+    node-ip: "{{ .NodeAddress }}"
+discovery:
+  bootstrapToken:
+    apiServerEndpoint: "{{ .JoinAddress }}"
```

Comment:
for v1alpha3 it's different:
https://github.com/kubernetes/kubernetes/blob/release-1.12/cmd/kubeadm/app/apis/kubeadm/v1alpha3/types.go#L299-L302

(thus the recommendation to have separate constants)

for v1beta* this seems OK.